File: python_sdk/src/reality_capture/specifications/eval_sortho.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class EvalSOrthoInputs(BaseModel):
    reference: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                       "pointing to segmented photos reference")
    prediction: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                        "pointing to segmented photos prediction")


class EvalSOrthoOutputs(BaseModel):
    report: Optional[str] = Field(None,
                                  description="Path to JSON report file with confusion matrix (bucket path in cloud, local path on-premise)")
    segmented_photos: Optional[str] = Field(None, alias="segmentedPhotos",
                                            description="Reality data ID (cloud) or local path (on-premise) of segmented photos, "
                                                        "annotated with confusion matrix index")
    segmentation2d: Optional[str] = Field(None, alias="segmentation2D",
                                          description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                                      "pointing to segmented photos")


class EvalSOrthoOutputsCreate(Enum):
    REPORT = "report"
    SEGMENTED_PHOTOS = "segmentedPhotos"
    SEGMENTATION2D = "segmentation2D"


class EvalSOrthoSpecificationsCreate(BaseModel):
    inputs: EvalSOrthoInputs = Field(description="Inputs")
    outputs: list[EvalSOrthoOutputsCreate] = Field(description="Outputs")


class EvalSOrthoSpecifications(BaseModel):
    inputs: EvalSOrthoInputs = Field(description="Inputs")
    outputs: EvalSOrthoOutputs = Field(description="Outputs")

File: typescript_sdk/reality_capture/src/specifications/eval_sortho.ts
/*import { z } from "zod";

export enum EvalSOrthoOutputsCreate {
  REPORT = "report",
  SEGMENTED_PHOTOS = "segmentedPhotos",
  SEGMENTATION2D = "segmentation2D",
}

export const EvalSOrthoInputsSchema = z.object({
  reference: z.string().describe(
    "Reality data id of ContextScene, pointing to segmented photos reference"
  ),
  prediction: z.string().describe(
    "Reality data id of ContextScene, pointing to segmented photos prediction"
  ),
});
export type EvalSOrthoInputs = z.infer<typeof EvalSOrthoInputsSchema>;

export const EvalSOrthoOutputsSchema = z.object({
  report: z
    .string()
    .regex(/^bkt:.+$/)
    .describe("Path in Bucket of json report with confusion matrix")
    .optional(),
  segmentedPhotos: z
    .string()
    .describe(
      "Reality data id of segmented photos, annotated with confusion matrix index"
    )
    .optional(),
  segmentation2D: z
    .string()
    .describe(
      "Reality data id of ContextScene, pointing to segmented photos"
    )
    .optional(),
});
export type EvalSOrthoOutputs = z.infer<typeof EvalSOrthoOutputsSchema>;

export const EvalSOrthoSpecificationsCreateSchema = z.object({
  inputs: EvalSOrthoInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(EvalSOrthoOutputsCreate)).describe("Outputs"),
});
export type EvalSOrthoSpecificationsCreate = z.infer<typeof EvalSOrthoSpecificationsCreateSchema>;

export const EvalSOrthoSpecificationsSchema = z.object({
  inputs: EvalSOrthoInputsSchema.describe("Inputs"),
  outputs: EvalSOrthoOutputsSchema.describe("Outputs"),
});
export type EvalSOrthoSpecifications = z.infer<typeof EvalSOrthoSpecificationsSchema>;*/